src/pyspiffe/spiffe_id/trust_domain.py

```diff
@@ -1,11 +1,59 @@
+from urllib.parse import urlparse
+
+from pyspiffe import SPIFFE_SCHEME
+
+EMPTY_DOMAIN_ERROR = 'Trust domain cannot be empty'
+SCHEMA_SUFFIX = '://'
+
+
 class TrustDomain(object):
     """
     Represents the name of a SPIFFE trust domain (e.g. 'domain.test').
 
-    :param trust_domain: The name of the Trust Domain
+    Args:
+        name(str): The name of the Trust Domain
+
+    Raises:
+        ValueError: is the name of the trust domain is empty, has a port, or contains an invalid scheme.
+
     """
 
-    name: str
+    def __init__(self, name: str):
+        self.__set_name(name)
+
+    def __str__(self):
+        return self.__name
+
+    def __eq__(self, other):
+        if isinstance(other, TrustDomain):
+            return self.__name == other.get_name()
+        return False
+
+    def __set_name(self, name):
+        if name == '' or name is None:
+            raise ValueError(EMPTY_DOMAIN_ERROR)
+
+        name = normalize(name)
+        uri = urlparse(name)
+        validate_uri(uri)
+        self.__name = uri.hostname.lower().strip()
+
+    def get_name(self):
+        return self.__name
+
+
+def normalize(name: str):
+    if not name.__contains__(SCHEMA_SUFFIX):
```

Comment:
what if we check that the name starts with the right schema?
```python
if not name.startswith(SPIFFE_SCHEME + SCHEME_SUFFIX):
```
We could also validate it's a valid url by parsing it, but that's probaby overkill.
Also, `__contains__`, as most of the magic methods, is not intended to be used directory. It's recommended to use `in` instead: `if not XXX in name`:

Reply:
Good catch! I replaced the magic method call by the `in` operator. The logic behind this is to add the default schema in case is doesn't have any and then parse the URL and performs the validations on the URL.

Reply:
sgtm